client_pool.go

```diff
@@ -0,0 +1,42 @@
+package cmhttp
+
+import (
+	"net/http"
+	"time"
+	"sync"
+
+	"github.com/bitly/go-hostpool"
+)
+
+// ClientPool creates a pool of HTTP clients that use an ε-greedy strategy to distribute
+// HTTP requests among multiple hosts. The pool transparently distributes the requests
+// among the hosts taking the individual request durations and failures into account.
+// A more detailed discussion of the underlying algorithm can be found at
+// https://godoc.org/github.com/bitly/go-hostpool#NewEpsilonGreedy
+//
+// See also https://en.wikipedia.org/wiki/Epsilon-greedy_strategy
+func ClientPool(hosts []string, decayDuration time.Duration, valueCalculator hostpool.EpsilonValueCalculator) Decorator {
+	pool := hostpool.NewEpsilonGreedy(hosts, decayDuration, valueCalculator)
+	clients := make(map[string]Client)
+	mux := &sync.Mutex{}
```

Comment:
Can we call this mu or lock instead? I find this confusing because of http.ServeMux.